mac_internet_sharing/__main__.py

```diff
@@ -17,6 +18,67 @@
 logger = logging.getLogger(__name__)
 
 
+async def plug_n_share_task(network_service_name: Optional[str], timeout: int = 5):
+    """ Continuously monitor for USB device changes and update SharingDevices. """
+    network_service = get_network_service(network_service_name)
+
+    logger.info('Starting USB device monitoring...')
+    prev_usb_devices = get_apple_usb_ethernet_interfaces()
+    logger.info(f'Connected devices: {set(prev_usb_devices.keys())}')
+
+    # Configure sharing with the initially detected interfaces.
+    configure(network_service, list(prev_usb_devices.values()))
+    await set_sharing_state(SharingState.ON)
+    while True:
+        current_usb_devices = get_apple_usb_ethernet_interfaces()
+        if prev_usb_devices != current_usb_devices:
+            if len(prev_usb_devices) > len(current_usb_devices):
+                logger.info(f'Removing devices {set(prev_usb_devices.keys()) - set(current_usb_devices.keys())}')
+            else:
+                logger.info(f'Adding devices {set(current_usb_devices.keys()) - set(prev_usb_devices.keys())}')
+            prev_usb_devices = current_usb_devices
+            interfaces = set(prev_usb_devices.values()).intersection(prev_usb_devices.values())
+            update_sharing_devices(interfaces)
+        await asyncio.sleep(timeout)
```

Comment:
Each time the list is updated we should print the entry we removed or added. For example the IP that became un/available. This should also be in the final commit and not this one